src/Framework/Constraint/StringContains.php

```diff
@@ -66,9 +66,21 @@ protected function matches($other): bool
         }
 
         if ($this->ignoreCase) {
+            /*
+             * We must use the multi byte safe version so we can accurately compare non latin upper characters with
+             * their lowercase equivalents.
+             */
             return \mb_stripos($other, $this->string) !== false;
```

Comment:
To speed the case-insensitive comparison up we could use `$other = mb_strtolower($other)` (as used in line 45) here to convert the searched phrase to lower case which is much faster here than using `mb_stripos` and then do the comparison binary safe in line 84.

Reply:
Feel free to send a pull request for this.